Source/WebCore/page/ResizeObserver.cpp

```diff
@@ -196,4 +222,23 @@ bool ResizeObserver::removeObservation(const Element& target)
     });
 }
 
+void ResizeObserver::CallbackForContainIntrinsicSize::onResize(const Vector<Ref<ResizeObserverEntry>>& entries)
+{
+    for (auto& entry : entries) {
+        if (auto* target = entry->target()) {
+            if (!target->isConnected()) {
+                target->removeLastRememberedSize();
+                target->disconnectFromResizeObservers();
+                continue;
+            }
+
+            if (auto* box = target->renderBox(); box && box->shouldApplySizeContainment())
+                continue;
+
+            auto contentBoxeSize = entry->contentBoxSize().at(0);
+            target->saveLastRememberedSize(ResizeObserverSize::create(contentBoxeSize->inlineSize(), contentBoxeSize->blockSize()));
```

Comment:
Maybe only one of `contain-intrinsic-width` and `contain-intrinsic-height` has the `auto` value. The last remembered size should be tracked independently for each axis (block and inline), see https://github.com/w3c/csswg-drafts/issues/7529. That's why test `auto-009.html` is still failing.

Can be done in a follow-up, just FYI.

Reply:
@Loirooriol Thanks a lot!
Yes, let me fix it in the follow-up patch:)